todo_app/test_integration.py

```diff
@@ -0,0 +1,49 @@
+import os
+import pytest
+import requests
+from todo_app import app
+from dotenv import load_dotenv, find_dotenv
+
+@pytest.fixture
+def client():
+    # Use our test integration config instead of the 'real' version
+    file_path = find_dotenv('.env.test')
+    load_dotenv(file_path, override=True)
+
+    # Create the new app.
+    test_app = app.create_app()
+
+    # Use the app to create a test_client that can be used in our tests.
+    with test_app.test_client() as client:
+        yield client
+
+class StubResponse():
+    def __init__(self, fake_response_data):
+        self.fake_response_data = fake_response_data
+
+    def json(self):
+        return self.fake_response_data
+
+# Stub replacement for requests.get(url)
+def stub(url, params):
+    test_board_id = os.environ.get('TRELLO_BOARD_ID')
+    fake_response_data = None
+    if url == f'https://api.trello.com/1/boards/{test_board_id}/lists':
+        fake_response_data = [{
+            'id': '123abc',
+            'name': 'To Do',
+            'cards': [{'id': '456', 'name': 'Test card'}]
+        }]
+        return StubResponse(fake_response_data)
+
+    raise Exception(f'Integration test stub no mock for url "{url}"')
+
+
+    # Replace requests.get(url) with our own function
+    monkeypatch.setattr(requests, 'get', stub)
+
+    # Make a request to our app's index page
+    response = client.get('/')
+
+    assert response.status_code == 200
+    assert 'Test card' in response.data.decode()
```

Comment:
You're missing defining an actual test! It must be a function called `test_` something for pytest to pick it up.
```suggestion
def test_integration(monkeypatch, client):
    # Replace requests.get(url) with our own function
    monkeypatch.setattr(requests, 'get', stub)

    # Make a request to our app's index page
    response = client.get('/')

    assert response.status_code == 200
    assert 'Test card' in response.data.decode()
```